media-gfx/zbar/zbar-0.23.ebuild

```diff
@@ -0,0 +1,203 @@
+
+LICENSE="LGPL-2.1"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~x86"
+
+IUSE="dbus graphicsmagick gtk gtk2 imagemagick introspection java jpeg png python qt5 static-libs test +threads v4l X xv"
+REQUIRED_USE="
+	introspection? (
+		|| ( gtk gtk2 )
+	)
+	python? ( ${PYTHON_REQUIRED_USE} )
+	qt5? ( X )
+	test? (
+		|| ( graphicsmagick imagemagick )
+		${PYTHON_REQUIRED_USE}
+		X png
+	)"
+
+RESTRICT="!test? ( test )"
+
+COMMON_DEPEND="
+	dbus? ( sys-apps/dbus[${MULTILIB_USEDEP}] )
+	gtk? (
+		dev-libs/glib:2[${MULTILIB_USEDEP}]
+		x11-libs/gtk+:3[${MULTILIB_USEDEP}]
+	)
+	gtk2? (
+		dev-libs/glib:2[${MULTILIB_USEDEP}]
+		x11-libs/gtk+:2[${MULTILIB_USEDEP}]
+	)
+	graphicsmagick? ( media-gfx/graphicsmagick:= )
+	imagemagick? ( media-gfx/imagemagick:= )
+	introspection? ( dev-libs/gobject-introspection )
+	jpeg? ( virtual/jpeg:0[${MULTILIB_USEDEP}] )
+	png? (
+		graphicsmagick? ( media-gfx/graphicsmagick:=[png] )
+		imagemagick? ( media-gfx/imagemagick:=[png] )
+	)
+	python? ( ${PYTHON_DEPS} )
+	qt5? (
+		dev-qt/qtcore:5
+		dev-qt/qtgui:5
+		dev-qt/qtwidgets:5
+		dev-qt/qtx11extras:5
+	)
+	v4l? ( media-libs/libv4l:0=[${MULTILIB_USEDEP}] )
+	X? (
+		x11-libs/libX11[${MULTILIB_USEDEP}]
+		x11-libs/libXext[${MULTILIB_USEDEP}]
+		xv? ( x11-libs/libXv[${MULTILIB_USEDEP}] )
+	)"
+
+RDEPEND="${COMMON_DEPEND}
+	java? ( virtual/jre:= )"
+
+DEPEND="${COMMON_DEPEND}
+	java? (
+		test? (
+			dev-java/junit:4
+			dev-java/hamcrest-core:1.3
+		)
+		>=virtual/jdk-1.8:=
+	)
+	test? (
+		${PYTHON_DEPS}
+		dev-python/pillow[${PYTHON_USEDEP}]
+	)"
+
+BDEPEND="
+	gtk? ( dev-util/glib-utils )
+	gtk2? ( dev-util/glib-utils )
+	sys-devel/gettext
+	virtual/pkgconfig"
+
+PATCHES=(
+	"${FILESDIR}/${PN}-0.10-errors.patch"
+	"${FILESDIR}/${P}-fix_python_detect.patch"
+)
+
+pkg_setup() {
+	if use python || use test; then
+		python-single-r1_pkg_setup
+	fi
+	java-pkg-opt-2_pkg_setup
+}
+
+src_prepare() {
+	default
+
+	if use python || use test; then
+		if use test; then
+			# make tests happy
+			# because one of the test requires loadable py module from the current ${BUILD_DIR}
+			sed -i \
+				-e "s|PYTHONPATH=@abs_top_srcdir@|PYTHONPATH=@builddir@|g" \
+				test/Makefile.am.inc || die
+		fi
+
+		python_fix_shebang \
+			examples/*.py \
+			test/{test_python,barcodetest}.py # test_pygtk.py — py2 only
+	fi
+
+	if use java; then
+		java-pkg-opt-2_src_prepare
+		sed "s|javadir = \$(pkgdatadir)|javadir = /usr/$(get_libdir)/zbar|" \
+			-i java/Makefile.am || die
+	fi
+
+	eautoreconf
+}
+
+multilib_src_configure() {
+	append-cppflags -DNDEBUG
+
+	local myeconfargs=(
+		$(use_with dbus)
+		$(use_with jpeg)
+		$(multilib_native_use_enable introspection)
+		$(multilib_native_use_with graphicsmagick)
+		$(multilib_native_use_with imagemagick)
+		$(multilib_native_use_with java)
+		$(multilib_native_use_with python)
+		$(use_enable static-libs static)
+		$(use_enable threads pthread)
+		$(use_enable v4l video) # using the 'qt5' flag is reqs to enable the 'X' also for building GUI version
```

Comment:
this comment looks strangely out of place, I would expect it either below where you enable qt5 or on top of REQUIRED_USE.